src/plugins/js-eval/run.js

```diff
@@ -78,18 +76,25 @@ const run = async (code, environment, timeout) => {
     });
   }
   if (environment === 'engine262') {
-    if (!runWithEngine262) {
-      const { Realm, initializeAgent, FEATURES, inspect } = require('engine262');
+    const { Agent, ManagedRealm, Value, CreateDataProperty, FEATURES, setSurroundingAgent, inspect: _inspect } = require('engine262');
 
-      initializeAgent({
-        features: FEATURES.map((o) => o.name),
-      });
+    const agent = new Agent({
+      features: FEATURES.map((o) => o.name),
+    });
+    setSurroundingAgent(agent);
 
-      const realm = new Realm();
+    const realm = new ManagedRealm();
 
-      runWithEngine262 = (code) => inspect(realm.evaluateScript(code), realm);
-    }
-    return runWithEngine262(code);
+    return new Promise((resolve, reject) => {
+      realm.scope(() => {
+        // const print = new Value((args) => {
+        //   console.log(...args.map((tmp) => _inspect(tmp)));
+        //   return Value.undefined;
+        // });
+        // CreateDataProperty(realm.GlobalObject, new Value('print'), print);
+        resolve(_inspect(realm.evaluateScript(code)));
+      });
+    });
```

Comment:
I'm curious about the failure cases here. Could you add tests for a syntax error, and runtime error with engine262?

Reply:
it's not ideal https://github.com/brigand/jellobot/pull/45/commits/e3ae50829c6df9d96ae654af2005324d23e414d9 because I don't know how t differentiate errors from `realm.evaluateScript(code)`

Reply:
Ok, it looks better now: https://github.com/brigand/jellobot/pull/45/commits/dca7f53452eb3535e02476e3264ce8b204a12d63